src/DuneFormatter.ml

```diff
@@ -1,6 +1,7 @@
 open Import
 
-let format (document : TextDocument.t) _options _token =
+let getFormatter toolchain =
+ fun [@bs] (document : TextDocument.t) _options _token ->
```

Comment:
What does the `[@bs]` annotation do here?

Reply:
It guarantees that the resulting function is uncurried from the JS side, which is what `registerDocumentFormattingEditProvider` expects. It's possible that it works fine without the annotation, but I don't want to take any chances.

https://bucklescript.github.io/docs/en/function#solution-guaranteed-uncurrying